src/java/htsjdk/samtools/SAMSequenceDictionary.java

```diff
@@ -163,8 +165,72 @@ public boolean equals(Object o) {
         return true;
     }
 
+    /** return a MD5 sum for ths dictionary, the checksum is not re-computed each time this
+     * method is called. <pre>md5( (seq1.name+seq1.length) + ' '+(seq2.name+seq2.length) + ' '+...)</pre>
+     * @return a MD5 checksum for this dictionary or the empty string if it is empty
+     */
+    public String md5() {
+        if(isEmpty()) return "";
+        try {
+            MessageDigest md5 = MessageDigest.getInstance("MD5");
+            md5.reset();
+            for(SAMSequenceRecord ssr : mSequences)  {
+                if(ssr.getSequenceIndex()>0)  md5.update((byte)' ');
+                md5.update(ssr.getSequenceName().getBytes());
+                md5.update(String.valueOf(ssr.getSequenceLength()).getBytes());
+            }
+            String  hash = new BigInteger(1, md5.digest()).toString(16);
+            if (hash.length() != 32) {
+                final String zeros = "00000000000000000000000000000000";
+                hash = zeros.substring(0, 32 - hash.length()) + hash;
+            }
+            return hash;
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+    
     @Override
     public int hashCode() {
         return mSequences.hashCode();
     }
+    
+    @Override
+    public String toString() {
+        return mSequences.toString();
```

Comment:
So this would return the class name + the hashcode of the array?  Perhaps it is more useful to return the md5?  What's the use case for this?